hudi-common/src/main/java/org/apache/hudi/exception/MissingSchemaFieldException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.exception;
+
+import java.util.List;
+
+/**
+ * Thrown when the schema of the incoming data is missing fields that are in the table schema.
+ */
+public class MissingSchemaFieldException extends SchemaCompatibilityException {
```

Comment:
How are `HoodieIncompatibleSchemaException`, `IncompatibleSchemaException`, and `UnsupportedAvroTypeException` now relate to `SchemaCompatibilityException`?  Should they be consolidated too?

Reply:
yes, lets ensure we revisit them all and remove unnecessary ones. we should have SchemaCompatibilityException at the top, and we can have subclasses that extends from this class for specific exceptions.

Reply:
IncompatibleSchemaException and UnsupportedAvroTypeException are from porting spark code I believe.
I got rid of HoodieIncompatibleSchemaException because it was only used once and I think it should just be a HoodieSchemaException. 

I think SchemaCompatibilityException should just be HoodieSchemaException in a lot of cases, but if we want to do that it should be a separate pr. In this pr, I make SchemaCompatibilityException extend HoodieSchemaException so now all the schema exceptions will extend from a single class